### PR TITLE
new stats in ES + post prometheus data in s3

### PR DESCRIPTION
**Pull request by abvgedeika**
added new stats in ES:
results.stats_total.(op rate/Total errors)
test_details.prometheus_report
test_details.grafana_snapshot
+
upload prometheus data in s3 bucket: cloudius-jenkins-test.s3.amazonaws.com

### CODE REVIEW

**Comment by amoskong** [comment]
inspekt complain:

```
2.55s$ inspekt style
PEP8 disabled: E501,E265,W601,E402
/home/travis/build/scylladb/scylla-cluster-tests/performance_regression_test.py:156:18: E127 continuation line over-indented for visual indent
PEP8 check fail: /home/travis/build/scylladb/scylla-cluster-tests/performance_regression_test.py
PEP8 compliance FAIL
The command "inspekt style" exited with 1.
```

File: sdcm/tester.py [hunk 1]
```diff
@@ -762,6 +757,33 @@ def clean_resources(self):
         if self.monitors is not None:
             self.monitors.get_backtraces()
             self.monitors.download_monitor_data()
+
+            # upload prometheus data
+            prometheus_folder = glob.glob(os.path.join(self.logdir, '*monitor*'))[0]
+            file_path = os.path.join(self.logdir, os.path.basename(os.path.normpath(self.job.logdir)))
```
Comment (amoskong) [inline sdcm/tester.py]: You create the zip file insider self.logdir, how about compress monitor log to the top of self.logdir?

file_path = os.path.normpath(self.job.logdir)

File: sdcm/tester.py [hunk 2]
```diff
@@ -762,6 +757,33 @@ def clean_resources(self):
         if self.monitors is not None:
             self.monitors.get_backtraces()
             self.monitors.download_monitor_data()
+
+            # upload prometheus data
+            prometheus_folder = glob.glob(os.path.join(self.logdir, '*monitor*'))[0]
```
Comment (amoskong) [inline sdcm/tester.py]: The monitor log directory is matched, why not directly match prometheus directory?

prometheus_folder = glob.glob(os.path.join(self.logdir, '*monitor*/prometheus/'))[0]

File: sdcm/tester.py (hunk 1 above)
Comment (amoskong) [inline sdcm/tester.py]: You create the zip file insider self.logdir,  use the name of last directory name of self.logdir.
but it compress the content of one sub-directory of self.logdir.

How about use the name of the sub-directory as compress file name?
file_path = os.path.join(self.logdir, os.path.basename(os.path.normpath(prometheus_folder)))

File: sdcm/tester.py [hunk 3]
```diff
@@ -11,50 +11,45 @@
...
-from .cluster import UserRemoteCredentials
-from .cluster import GCECredentials
-
 from .cluster import ScyllaAWSCluster
+from .cluster import ScyllaGCECluster
 from .cluster import ScyllaLibvirtCluster
 from .cluster import ScyllaOpenStackCluster
-from .cluster import ScyllaGCECluster
-
+from .cluster import UserRemoteCredentials
```
Comment (amoskong) [inline sdcm/tester.py]: Can you use a split commit for the cleanup of import?

File: sdcm/tester.py [hunk 4]
```diff
@@ -762,6 +757,33 @@ def clean_resources(self):
...
+            grafana_snapshots = glob.glob(os.path.join(self.logdir, '*monitor*/*grafana-snapshot*'))
+            if grafana_snapshots:
+                result_path = '%s.png' % file_path
+                with open(grafana_snapshots[0]) as fh:
+                    mydata = fh.read()
+                    url_s3 = 'https://cloudius-jenkins-test.s3.amazonaws.com/%s/%s' % (
+                        os.environ.get('JOB_NAME', "local_run"), os.path.basename(
+                            os.path.normpath(result_path)))
+                    self.log.info("uploading grafana snapshot data on %s" % url_s3)
+                    response = requests.put(url_s3, data=mydata)
```
Comment (amoskong) [inline sdcm/tester.py]: url_s3 is enough? or url_s3 + '.png' ?

File: sdcm/tester.py [hunk 5]
```diff
@@ -762,6 +757,33 @@ def clean_resources(self):
...
+            file_path = os.path.join(self.logdir, os.path.basename(os.path.normpath(self.job.logdir)))
+            shutil.make_archive(file_path, 'zip', prometheus_folder)
+            result_path = '%s.zip' % file_path
+            with open(result_path) as fh:
+                mydata = fh.read()
+                url_s3 = 'https://cloudius-jenkins-test.s3.amazonaws.com/%s/%s' % (
+                    os.environ.get('JOB_NAME', "local_run"), os.path.basename(
+                        os.path.normpath(result_path)))
+                self.log.info("uploading prometheus data on %s" % url_s3)
+                response = requests.put(url_s3, data=mydata)
```
Comment (amoskong) [inline sdcm/tester.py]: url_s3 is enough? or url_s3 + '.zip' ?

File: performance_regression_test.py [hunk 6]
```diff
@@ -146,9 +146,16 @@ def generate_stats_json(self, results, cmds=[]):
...
             subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip()
 
-        if os.environ.get('JOB_NAME'):
-            metrics['test_details']['job_name'] = os.environ.get('JOB_NAME', '')
+        job_name = os.environ.get('JOB_NAME', 'local_run')
+        metrics['test_details']['job_name'] = job_name
+        if os.environ.get('BUILD_URL'):
             metrics['test_details']['job_url'] = os.environ.get('BUILD_URL', '')
+        if self.monitors:
+            zip_file = os.path.join(self.logdir, os.path.basename(os.path.normpath(self.job.logdir)))
```
Comment (amoskong) [inline performance_regression_test.py]: The way to get the url_s3 from self.logdir is specific, and it's used in many place of this patch, can we add a helper function to do this?

File: sdcm/tester.py (hunk 4 above)
Comment (abvgedeika) [inline sdcm/tester.py]: result_path = '%s.png' % file_path

File: sdcm/tester.py (hunk 5 above)
Comment (abvgedeika) [inline sdcm/tester.py]: result_path contains zip:
result_path = '%s.zip' % file_path

File: sdcm/tester.py (hunk 2 above)
Comment (abvgedeika) [inline sdcm/tester.py]: ok, will use only inner prometheus folder but code should be:
prometheus_folder = glob.glob(os.path.join(self.logdir, '*monitor*/*monitor*/prometheus/'))[0]

File: sdcm/tester.py (hunk 1 above)
Comment (abvgedeika) [inline sdcm/tester.py]: non very clean messages for me)
but okay, I create archive like job-2017-06-01T10.26-6ff133e.zip(with only prometheus folder) inside of self.job.logdir

Comment (amoskong) [inline sdcm/tester.py]: That's fine :-)

**Pull request merged**